test/nodes/test_web_retriever.py

```diff
@@ -1,163 +1,246 @@
-            self.text = text
-            self.status_code = status_code
-
-    def get(url, headers, timeout):
-        return MockResponse("mocked", 200)
-
-    def get_content(self, text: str) -> str:
-        return "What are the top solutions for\nArya Stark's Father\nWe found 1 solutions for\nArya Stark's Father\n.The top solutions is determined by popularity, ratings and frequency of searches. The most likely answer for the clue is NED..."
-
-    monkeypatch.setattr(WebSearch, "run", mock_web_search_run)
-    monkeypatch.setattr(ArticleExtractor, "get_content", get_content)
-    monkeypatch.setattr(requests, "get", get)
-
-    web_retriever = WebRetriever(api_key="", top_search_results=2, mode="raw_documents")
-    result = web_retriever.retrieve(query="Who is the father of Arya Stark?")
-    assert len(result) == 1
-    assert isinstance(result[0], Document)
-    assert (
-        result[0].content
-        == "What are the top solutions for\nArya Stark's Father\nWe found 1 solutions for\nArya Stark's Father\n.The top solutions is determined by popularity, ratings and frequency of searches. The most likely answer for the clue is NED..."
+def test_init_default_parameters():
+    retriever = WebRetriever(api_key="test_key")
+
+    assert retriever.top_k == 5
+    assert retriever.mode == "snippets"
+    assert retriever.preprocessor is None
+    assert retriever.cache_document_store is None
+    assert retriever.cache_index is None
+    assert retriever.cache_headers is None
+    assert retriever.cache_time == 1 * 24 * 60 * 60
+
+
+@pytest.mark.unit
+def test_init_custom_parameters():
+    preprocessor = PreProcessor()
+    document_store = MagicMock(spec=BaseDocumentStore)
+    headers = {"Test": "Header"}
+
+    retriever = WebRetriever(
+        api_key="test_key",
+        search_engine_provider="SerperDev",
+        top_search_results=15,
+        top_k=7,
+        mode="preprocessed_documents",
+        preprocessor=preprocessor,
+        cache_document_store=document_store,
+        cache_index="custom_index",
+        cache_headers=headers,
+        cache_time=2 * 24 * 60 * 60,
     )
-    assert result[0].score == None
-    assert result[0].meta["url"] == "https://crossword-solver.io/clue/arya-stark%27s-father/"
-    # Only preprocessed docs but not raw docs should have the _split_id field
-    assert "_split_id" not in result[0].meta
+
+    assert retriever.top_k == 7
+    assert retriever.mode == "preprocessed_documents"
+    assert retriever.preprocessor == preprocessor
+    assert retriever.cache_document_store == document_store
+    assert retriever.cache_index == "custom_index"
+    assert retriever.cache_headers == headers
+    assert retriever.cache_time == 2 * 24 * 60 * 60
+
+
+@pytest.mark.unit
+def test_retrieve_from_web_all_params(mock_web_search):
+    wr = WebRetriever(api_key="fake_key")
+
+    preprocessor = PreProcessor()
+
+    result = wr._retrieve_from_web("Who is the boyfriend of Olivia Wilde?", preprocessor)
```

Comment:
```suggestion
    result = wr._retrieve_from_web(query="Who is the boyfriend of Olivia Wilde?", preprocessor=preprocessor)
```
If possible, let's try to be explicit with parameter names.
(Also in the following tests...)